scripts/checkout.sh

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+
+set -ex
+
+checkout() {
+	local SRC="$1"
+	local REF="$2"
+	# if ref is branch or tag, retrieve its canonical form
+	local REF_CANON=$(git -C "$SRC" ls-remote --refs --heads --tags origin "$REF" | awk '{print $2}')
+	if [ -n "$REF_CANON" ]; then
+		# if branch or tag then create it locally too
+		FETCH_REF="$REF_CANON:$REF_CANON"
+	else
+		# otherwise use REF input as-is (e.g. commit hash)
+		REF_CANON="$REF"
+		FETCH_REF="$REF"
+	fi
+	git -C "$SRC" fetch --depth 1 origin "$FETCH_REF"
+	git -C "$SRC" checkout -q "$REF_CANON"
+}
+
+
+# Only execute checkout function above if this file is executed, not sourced from another script
+prog=checkout.sh # needs to be in sync with this file's name
+if [ "$(basename -- $0)" == "$prog" ]; then
```

Comment:
This check is not needed here, but i thought this file could be reused (in other scripts as well) as it's basically circumventing a git limitation. I wish this beahvior was just a flag on git fetch or that git clone accepted commit hashes.